File: Go/Easy/Bubble Sort/solution_1.go
package main

func BubbleSort(array []int) []int {

	sorted := false
	counter := 0

	for !sorted {

		sorted = true

		for i := 0; i < len(array)-1-counter; i++ {

			if array[i] > array[i+1] {
				array[i], array[i+1] = array[i+1], array[i]
				sorted = false
			}
		}
		counter++
	}
	return array
}

File: Python/Easy/Bubble Sort/solution_1.py
def bubbleSort(array):

    sorted = False
    while not sorted:
        sorted = True
        for idx in range(len(array) - 1):
            if array[idx] > array[idx + 1]:
                array[idx] , array[idx + 1] = array[idx + 1] , array[idx]
                sorted = False
                

    return array